### ISSUE TITLE
Variable explorer: add save as HDF5

### ISSUE THREAD

**Issue by spyder-issues-migrator** [spyder-ide/spyder#75]
_From [tim.mich...@gmail.com](https://code.google.com/u/117399810383525443135/) on 2009-10-11T07:06:19Z_

save workspace contents to a pytables file would be the arrays and
variables accessible and reusable for other software ourside Spyder.

_Original issue: http://code.google.com/p/spyderlib/issues/detail?id=75_

**Comment by spyder-issues-migrator** [comment]
_From [pierre.raybaut](https://code.google.com/u/pierre.raybaut/) on 2009-10-11T05:44:40Z_

You mean save as an HDF5 file.
I've already thought about it but I'm not sure I have the time to implement it in v1.0.

**Summary:** Import data from HDF5 files in Interactive console's workspace  
  **Status:** Accepted  
  **Labels:** -Type-Defect Type-Enhancement Milestone-Release1.1

**Comment by spyder-issues-migrator** [comment]
_From [pierre.raybaut](https://code.google.com/u/pierre.raybaut/) on 2009-10-11T05:45:37Z_

**Summary:** Interactive console's workspace: add save as HDF5

**Comment by spyder-issues-migrator** [comment]
_From [DavidAnt...@gmail.com](https://code.google.com/u/108698404730356683634/) on 2009-10-26T21:43:26Z_

This sounds like a good idea, however for compatibility with other programs, it is
better to use h5py instead of pytables, because pytables adds all this messy
meta-data which only it understands, whereas h5py creates raw hdf5 which is more
suitable to read with other programs.

**Comment by spyder-issues-migrator** [comment]
_From [pierre.raybaut](https://code.google.com/u/pierre.raybaut/) on 2010-02-06T07:45:42Z_

**Labels:** -Milestone-Release1.1 Milestone-Release1.2

**Comment by spyder-issues-migrator** [comment]
_From [pierre.raybaut](https://code.google.com/u/pierre.raybaut/) on 2010-02-06T07:50:53Z_

**Labels:** -Priority-Medium Priority-Low

**Comment by spyder-issues-migrator** [comment]
_From [pierre.raybaut](https://code.google.com/u/pierre.raybaut/) on 2010-06-19T05:45:46Z_

**Labels:** -Milestone-Release1.2 Milestone-Release2.0

**Comment by spyder-issues-migrator** [comment]
_From [pierre.raybaut](https://code.google.com/u/pierre.raybaut/) on 2010-06-19T05:46:05Z_

**Summary:** Variable explorer: add save as HDF5

**Comment by spyder-issues-migrator** [comment]
_From [pierre.raybaut](https://code.google.com/u/pierre.raybaut/) on 2010-10-17T12:58:28Z_

**Labels:** -Milestone-Release2.0 Milestone-Release2.1

**Comment by spyder-issues-migrator** [comment]
_From [DavidAnt...@gmail.com](https://code.google.com/u/108698404730356683634/) on 2010-11-04T18:23:22Z_

I had a quick look at the example file I/O plugin, and was able to knock together the attached hdf5 plugin quite quickly.  There are a few caveats, which are documented in the python file.

Unfortunately, importing h5py (or pytables when I tried that) causes Spyder to freeze.  This seems to be related to importing the module during the save dialog box.  Spyder has no trouble importing these modules from the command prompt.  I'm not sure what the solution is.

**Attachment:** [io_hdf5.py](http://code.google.com/p/spyderlib/issues/detail?id=75#hc9)

**Comment by spyder-issues-migrator** [comment]
_From [pierre.raybaut](https://code.google.com/u/pierre.raybaut/) on 2011-05-16T01:04:52Z_

**Labels:** Cat-VariableExplorer

**Comment by spyder-issues-migrator** [comment]
_From [pierre.raybaut](https://code.google.com/u/pierre.raybaut/) on 2011-05-24T11:40:55Z_

I took the time earlier today to find out why importing h5py was causing Spyder to freeze.

Apparently this is a combination of two problems:
1. h5py.__init__ is importing IPython, which is not good because I know for a fact that -due to IPython internals- IPython must not be imported before the session has been initialized within Spyder
2. Importing h5py in a separate thread in the remote process is blocking the process, for an unknown reason

I've implemented workarounds and it seems to be working in all situations.
